Engines/Wine/Engine/Versions/script.js

```diff
@@ -1,8 +1,139 @@
+        // check version format
+        const versionRegExp = /^(\d+\.\d+(\.\d+)?)(.*)?$/;
+        if (!versionRegExp.test(a.version)) {
+            throw new Error(`invalid Wine version "${a.version}`);
+        }
+        if (!versionRegExp.test(b.version)) {
+            throw new Error(`invalid Wine version "${a.version}`);
+        }
+
+        // sort
+        const aVersionParts = a.version.match(versionRegExp);
+        const bVersionParts = b.version.match(versionRegExp);
+
+        const aVersionNumbers = Array.from(aVersionParts[1].split('.')).map(item => Number(item));
+        const bVersionNumbers = Array.from(bVersionParts[1].split('.')).map(item => Number(item));
+
+        // ensure that cases where not all version numbers (major, minor, patch) are set are handled correctly
+        const maxVersionIdx = 2;
+        let i;
+        for (i = 0; i <= maxVersionIdx; i++) {
+            if (typeof aVersionNumbers[i] === 'undefined') {
+                aVersionNumbers[i] = 0;
+            }
+            if (typeof bVersionNumbers[i] === 'undefined') {
+                bVersionNumbers[i] = 0;
+            }
+        }
+
+        // ensure that cases where the description is not set are handled correctly
+        let aVersionDescription = aVersionParts[3];
+        if (typeof aVersionDescription === 'undefined') {
+            aVersionDescription = "";
+        }
+
+        let bVersionDescription = bVersionParts[3];
+        if (typeof bVersionDescription === 'undefined') {
+            bVersionDescription = "";
+        }
+
+        // major
+        if (aVersionNumbers[0] != bVersionNumbers[0]) {
+            return aVersionNumbers[0] - bVersionNumbers[0];
+        }
+        // minor
+        if (aVersionNumbers[1] != bVersionNumbers[1]) {
+            return aVersionNumbers[1] - bVersionNumbers[1];
+        }
+        // patch
+        if (aVersionNumbers[2] != bVersionNumbers[2]) {
+            return aVersionNumbers[2] - bVersionNumbers[2];
+        }
+        // description
+        if (aVersionDescription < bVersionDescription) {
+            return -1;
+        } else {
+            return 1;
+        }
+    }
+    );
+}
+
+/**
+ * Fetches the latest available Wine version based on the category and version regex
+ * @param {wizard} wizard setup wizard to show the download progress of the versions json
+ * @param {string} category (e.g. "upstream-linux-x86")
+ * @param {string} regex regex the version must fulfill
+ * @returns {void}
+ */
+function getLatestVersion(wizard, category, regex) {
+    const versionsJson = module.getAvailableVersions(wizard);
+
+    const packages = versionsJson
+        .filter(distribution => distribution.name === category)
+        .flatMap(distribution => distribution.packages);
+
+    const regExp = new RegExp(regex);
+    const versions = packages
+        .filter(({version}) => regExp.test(version))
+        .flatMap(package => package.version);
+
+    return versions[versions.length-1];
```

Comment:
Is it always the case that the last version is the "latest" version?

Reply:
I think so. Do you have any particular use case in mind?

Reply:
I don't have an use case in mind. The issue is just that this code allows us in theory to process engine version lists from multiple sources. Over some of these sources we have control, there we can ensure the ordering of the versions. Other sources we possibly know of but have no control. In addition there can also be a number of sources that we possibly don't even know of. 

For this reason I think we should add some additional logic or rules to ensure that the "latest" version is returned.

Reply:
I don't see how we could determine the latest version if this logic does not work.

Reply:
We can sort them before we take the first/last entry right?

```javascript 
const versions = ...

sortVersions(versions);

return versions[versions.length - 1];
```

Reply:
This is what I do already. The `getAvailableVersions` does the sorting.

Reply:
Then ignore this comment :)